scanner/scanner.go

```diff
@@ -0,0 +1,297 @@
+package scanner
+
+import (
+	"strconv"
+	"unicode"
+	"unicode/utf8"
+
+	"github.com/youla-dev/godenv/token"
+)
+
+const (
+	bom = 0xFEFF // byte order mark, only permitted as the first character
+	eof = -1     // eof indicates the end of the file.
+)
+
+// Scanner converts a sequence of characters into a sequence of tokens.
+type Scanner struct {
+	input      string
+	ch         rune // current character
+	prevOffset int  // position before current character
+	offset     int  // character offset
+	peekOffset int  // position after current character
+}
+
+// New returns new Scanner.
+func New(input string) *Scanner {
```

Comment:
It might be helpful to give an example of 'Scanner' usage

Reply:
I guess there is no way anyone will use `Scanner` without `Parser` 😅 Thus, there is no actual need for an example for it.

Reply:
Ok, this is a little bit of my paranoia, everything that is exported can be used against you.